include/roboteam_utils/Line.h

```diff
@@ -1,101 +1,126 @@
+     * @param v2 An arbitrary different point on the Line (make sure that it is different than the v2 point).
      */
-    Vector2 start;
+    explicit Line(const Vector2 &v1, const Vector2 &v2);
+
     /**
-     * @brief End of the line
+     * Create a new Line from a LineSegment by infinitely expanding that LineSegment.
      *
+     * @param other LineSegment that is infinitely expanded. Make sure that the LineSegment has a length > 0.
      */
-    Vector2 end;
+    explicit Line(const LineSegment &other) noexcept;
+
     /**
-     * @brief Construct a new Line object
+     * Computes the Euclidean distance from point to the Line. The theory behind this algorithm is explained in: http://www.randygaul.net/2014/07/23/distance-point-to-line-segment/
      *
+     * @param point Point to which the distance is computed.
+     * @return A double value >= 0.0 which represents the distance to this Line.
      */
-    constexpr Line() = default;
+    [[nodiscard]] double distanceToLine(const Vector2 &point) const;
 
     /**
-     * @brief Constructs a new Line object
-     *
-     * @param _start Start of the Line
-     * @param _end End of the Line
+     * Projects the point onto this Line, i.e. find the location on this Line closest to that point.
      *
+     * @param point Point that is projected on this Line.
+     * @return The projection point. Note that this point is located on this Line and it is orthogonal to the given point.
      */
-    constexpr Line(const Vector2 &_start, const Vector2 &_end) noexcept: start{_start}, end{_end} {};
-    /**
-     * @brief Constructs a new Line from a LineSegment.
-     * @param other LineSegment to use for construction
-     */
-    explicit Line(const LineSegment &other) noexcept;
+    [[nodiscard]] Vector2 project(const Vector2 &point) const;
+
     /**
-     * @brief Gets the length of the vector representation of this line.
-     *        Note in this case it does not really make sense as this class represents lines of infinite length
-     * Literally:
-     *      (end - start).length()
-     * @return double Gets the length of the line
+     * Get the intersection point between two (infinite) Lines. In case there is no intersection (Lines are parallel and distinct), return std
+     *
+     * @param line The other (infinite) Line.
+     * @return In case there is no intersection (i.e. the Lines are parallel and distinct), return std::nullopt. In case there is a single intersection, return that point (Lines
+     * are non-parallel). In case there are multiple (infinitely many) intersections, return the intersection point closest to the origin (0,0).
      */
-    [[nodiscard]] double length() const;
+    [[nodiscard]] std::optional<Vector2> intersect(const Line &line) const;
 
     /**
-     * @brief Gets the length of the vector representation of this line
+     * Determine whether a point lies on this Line. This function is NOT protected against double/float rounding issues.
```

Comment:
Why not?

Reply:
Forgot to update the documentation. It should be fixed in the other PR (#79)